### DISCUSSION TITLE
How to set slide width in virtual mode?

### DISCUSSION THREAD

**Discussion by zhaozhemin** [nolimits4web/swiper#6217] [Q&A]
Normally, you can set slide width by writing CSS, e.g. `.swiper-slide { width: 80% }`, when `slidesPerView` is `auto`. But `virtual: true` cannot be used with `slidesPerView: auto`. If I override the width with `!important` using CSS, the width is right but the `transform` etc., is all wrong.

**Comment by danilosimonatto** [comment]
You can't use slidesPerView in auto because as stated in #2287 virtual slides must know the slide size before it is rendered.
